Add many functionalities for folders and notes

SharpBit opened the pull request:
### Notes
- Rename 
- Move
- Delete
### Folders
- Move

Also fixes the bug when the folder that the currently opened note is in or the currently opened note is renamed, moved, or deleted, and then you save it. It now will prompt a Save As since the note no longer "exists" (it may exist if it was moved or renamed, just not at the old path).

SharpBit committed:
add functionality to move folders and notes

SharpBit committed:
add ability to rename/delete notes and move folders/notes

SharpBit committed:
Merge branch 'main' into move_items

SharpBit committed:
fix comments again since it keeps getting changed back

standard-affect commented on src/navigation/menu_bar.py:
```diff
@@ -462,6 +536,9 @@ def _save_file_at_path(self, path: str, text: str) -> None:
 
+    def validate_path(self, path: str) -> bool:
+        """Validates if the path can be created."""
+
```
This function is empty - did you start it but end up not needing it?

standard-affect reviewed (commented):
Renaming works, though I can't get moving to work. At any rate, a lot of good code done quickly.

CupOfGeo commented:
Again i get this when making a new folder but if i click ok it goes away same with the other comment i had
![image](https://user-images.githubusercontent.com/23179046/125998049-1b82ef26-5773-4fc7-a444-97ea4fe0326b.png)

CupOfGeo commented:
I also got this error while trying to cancel honestly it was really confusing trying to move my poop.txt to my new toilet folder 
Unhandled exception in event loop:
  File "/Users/mazzeogeorge/Code/adaptable-antelopes/src/navigation/menu_bar.py", line 223, in coroutine
    shutil.move(item_path, move_path)
  File "/usr/local/Cellar/python@3.8/3.8.10/Frameworks/Python.framework/Versions/3.8/lib/python3.8/shutil.py", line 780, in move
    if os.path.isdir(dst):
  File "/usr/local/Cellar/python@3.8/3.8.10/Frameworks/Python.framework/Versions/3.8/lib/python3.8/genericpath.py", line 42, in isdir
    st = os.stat(s)

Exception stat: path should be string, bytes, os.PathLike or integer, not NoneType
Press ENTER to continue...

SharpBit committed:
fix some bugs

- make it clear if something already exists when moving a file
- make it so you cant move the root directory
- fix error when cancelling move file

SharpBit committed:
remove unnecessary func

SharpBit committed:
Skip the scroll menu if there are no folders yet (to not confuse users)

CupOfGeo commented:
Trying to delete a subfolder

Unhandled exception in event loop:
  File "/Users/mazzeogeorge/Code/adaptable-antelopes/src/navigation/menu_bar.py", line 444, in coroutine
    if self.application_state.current_path.startswith(path):

Exception 'NoneType' object has no attribute 'startswith'
Press ENTER to continue...

SharpBit committed:
fix bug when deleting/moving/renaming and a new file is open

CupOfGeo commented on src/navigation/menu_bar.py:
```diff
@@ -197,22 +198,85 @@ def do_new_file(self) -> None:
+                path = await self.show_dialog_as_float(dialog)
+                if not path:
+                    return
+            else:
```
rad 🤙
